### PR TITLE
DashR HTML Components

### PR DESCRIPTION
**Pull request by KPhans**
Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [ ] I understand

### CODE REVIEW

File: dashr/chapters/dash-html-components/index.R [hunk 1]
```diff
@@ -13,6 +13,7 @@ Instead of writing HTML or using an HTML templating engine, you compose your lay
 The source for this library is on GitHub: [plotly/dash-html-components](https://github.com/plotly/dash-html-components).
 Here is an example of a simple HTML structure:
   "),
+
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: @KPhans The magnifying glass icon for the search bar at upper right is rendering as a "fl" ligature; we should fix this, not sure what's happening there.

File: dashr/chapters/dash-html-components/index.R [hunk 2]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+             Dash's [core component library](/dash-core-components) also supports
+             [Markdown](http://commonmark.org/help).
+             "),
+
+dccMarkdown("
+    ```r
+    library(dashHtmlComponents)
+    dccMarkdown(\'\'\'
+    #### Dash and Markdown
+    Dash supports [Markdown](http://commonmark.org/help).
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Let's replace this URL for CommonMark with the updated one for `react-markdown`; this is an issue with the Python docs as well, so I'll open a PR to update the URL there too.

https://daringfireball.net/projects/markdown/

File: dashr/chapters/dash-html-components/index.R [hunk 3]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+    It includes a syntax for things like **bold text** and *italics*,
+    [links](http://commonmark.org/help), inline `code` snippets, lists,
+    quotes, and more.
+    \'\'\')
+    ```
+    "),
+
+dccMarkdown("
+    #### Dash and Markdown
+    Dash supports [Markdown](http://commonmark.org/help).
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Same, replace URL with https://daringfireball.net/projects/markdown/

File: dashr/chapters/dash-html-components/index.R [hunk 4]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+    Markdown is a simple way to write and format text.
+    It includes a syntax for things like **bold text** and *italics*,
+    [links](http://commonmark.org/help), inline `code` snippets, lists,
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: https://daringfireball.net/projects/markdown/

File: dashr/chapters/dash-html-components/index.R [hunk 5]
```diff
@@ -13,6 +13,7 @@ Instead of writing HTML or using an HTML templating engine, you compose your lay
 The source for this library is on GitHub: [plotly/dash-html-components](https://github.com/plotly/dash-html-components).
 Here is an example of a simple HTML structure:
   "),
+  
   utils$LoadAndDisplayComponent(
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Looks like a missing apostrophe between "Dash" and "s", and then some stray backpacks in the sample code:

Screenshot 2019-06-28 17 15 48

File: dashr/chapters/dash-html-components/index.R [hunk 6]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+    [links](http://commonmark.org/help), inline `code` snippets, lists,
+    quotes, and more.", className='example-container'),
+
+dccMarkdown("
+If you're using HTML components, then you also have access to
+properties like `style`, `class`, and `id`.
+All of these attributes are available in the Python classes.
+The HTML elements and Dash classes are mostly the same but there are
+a few key differences:
+- The `style` property is a dictionary
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Replace "dictionary" with "named list"

File: dashr/chapters/dash-html-components/index.R [hunk 7]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+    Dash supports [Markdown](http://commonmark.org/help).
+    Markdown is a simple way to write and format text.
+    It includes a syntax for things like **bold text** and *italics*,
+    [links](http://commonmark.org/help), inline `code` snippets, lists,
+    quotes, and more.", className='example-container'),
+
+dccMarkdown("
+If you're using HTML components, then you also have access to
+properties like `style`, `class`, and `id`.
+All of these attributes are available in the Python classes.
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Replace "Python classes" with "R functions"

File: dashr/chapters/dash-html-components/index.R [hunk 8]
```diff
@@ -26,8 +27,74 @@ htmlDiv(list(
...
+The HTML elements and Dash classes are mostly the same but there are
+a few key differences:
+- The `style` property is a dictionary
+- Properties in the `style` dictionary are camelCased
+- The `class` key is renamed as `className`
+- Style properties in pixel units can be supplied as just numbers without the `px` unit
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: Need a linefeed between "px unit" and the following line

File: dashr/chapters/dash-html-components/index.R [hunk 9]
```diff
@@ -13,6 +13,7 @@ Instead of writing HTML or using an HTML templating engine, you compose your lay
 The source for this library is on GitHub: [plotly/dash-html-components](https://github.com/plotly/dash-html-components).
```
Comment (rpkyle) [inline dashr/chapters/dash-html-components/index.R]: As we discussed, "using R structures" sounds odd, let's change it to "...compose your layout using R with the..."

**Comment by KPhans** [comment]
@rpkyle please see latest commit for this PR! should be good to go

**Pull request merged**